numpy/f2py/cfuncs.py

```diff
@@ -649,6 +649,7 @@
         STRINGCOPYN(buf, str, n);
         return 1;
     }
+    return -1;
```

Comment:
This only sort of works for the specific case where `None` is to be passed.

See the discussion in https://github.com/numpy/numpy/issues/24662#issuecomment-1818002692.